mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+from mlflow.exceptions import MlflowException
+from mlflow.models import Model
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.models.signature import ModelSignature
+from mlflow.models.utils import ModelInputExample, _save_example
+from mlflow.spark import _mlflowdbfs_path, _should_use_mlflowdbfs, _maybe_save_model, _HadoopFileSystem
+from mlflow.store.artifact.databricks_artifact_repo import DatabricksArtifactRepository
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.tracking.artifact_utils import (
+    _download_artifact_from_uri,
+    _get_root_uri_and_artifact_path,
+)
+from mlflow.utils import databricks_utils
+from mlflow.utils.autologging_utils import autologging_integration, safe_patch
+from mlflow.utils.docstring_utils import format_docstring, LOG_MODEL_PARAM_DOCS
+from mlflow.utils.environment import (
+    _mlflow_conda_env,
+    _process_pip_requirements,
+    _process_conda_env,
+    _CONDA_ENV_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+    _CONSTRAINTS_FILE_NAME,
+    _PYTHON_ENV_FILE_NAME,
+    _PythonEnv,
+)
+from mlflow.utils.file_utils import TempDir, write_to, shutil_copytree_without_file_permissions
+from mlflow.utils.model_utils import (
+    _get_flavor_configuration_from_uri,
+    _validate_and_copy_code_paths,
+    _add_code_from_conf_to_system_path,
+)
+from mlflow.utils.uri import (
+    is_local_uri,
+    append_to_uri_path,
+    dbfs_hdfs_uri_to_fuse_path,
+    is_valid_dbfs_uri,
+    get_databricks_profile_uri_from_artifact_uri,
+    generate_tmp_dfs_path,
+)
+
+def validate_env_vars():
+    if not all(os.environ.get(var) for var in JOHNSNOWLABS_ENV_VARS):
+        raise Exception(
+            'Please set SECRET environment variable to your secret key before saving or logging your model')
+
+FLAVOR_NAME = "johnsnowlabs"
+JOHNSNOWLABS_ENV_VARS = ('SECRET', 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'SPARK_NLP_LICENSE')
+
+
+def download_url(url, save_path):
+    file_name = url.split("/")[-1]
+    save_path = os.path.join(save_path, file_name)
+    print(f"Downloading {file_name}")
+    # Download the file from `url` and save it locally under `file_name`:
+    with urlopen(url) as response, open(save_path, "wb") as out_file:
+        shutil.copyfileobj(response, out_file)
+
+
+def download_sparknlp_jar(out_folder):
+    SPARK_NLP_JAR = f'spark-nlp-assembly-{settings.raw_version_nlp}.jar'
+    download_url(
+        f"https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/{SPARK_NLP_JAR}",
+        out_folder)
+
+
+def download_sparknlp_jsl_jar(out_folder, secret):
+    SPARK_NLP_INTERNAL_JAR = f'spark-nlp-jsl-{settings.raw_version_medical}.jar'
+    download_url(
+        f"https://pypi.johnsnowlabs.com/{secret}/{SPARK_NLP_INTERNAL_JAR}",
+        out_folder)
+
+
+HEALTHCARE_wheel_URI = 'https://pypi.johnsnowlabs.com/{secret}/spark-nlp-jsl/spark_nlp_jsl-{SPARK_NLP_INTERNAL_VERSION}-py3-none-any.whl'
```

Comment:
Please capitalize constants and move to top of module

Reply:
thanks updated as suggested